src/python/pants/bsp/util_rules/targets.py

```diff
@@ -524,9 +538,22 @@ class DependencySourcesHandlerMapping(BSPHandlerMapping):
 
 @rule
 async def bsp_dependency_sources(request: DependencySourcesParams) -> DependencySourcesResult:
-    # TODO: This is a stub.
+    dependency_modules = await Get(
+        DependencyModulesResult, DependencyModulesParams, DependencyModulesParams(request.targets)
+    )
```

Comment:
This would have its own backend-specific rule I suppose, for now it just pulling it from `dependency_modules` fully expecting it to be a maven module, again to try it out without implementing a whole rule-indirection for it

Reply:
Understood. Maybe add a TODO comment as a reminder to create the union hook which makes this backend-agnostic?

Reply:
> This would have its own backend-specific rule I suppose

Yes, because the core BSP logic should not be aware of the per-language specifics.